miniapps/dpg/util/blockstaticcond.hpp

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_BLOCK_STATIC_CONDENSATION
+#define MFEM_BLOCK_STATIC_CONDENSATION
+
+#include "mfem.hpp"
+
+namespace mfem
+{
+
+class BlockStaticCondensation
+{
+   int height, width;
+   int nblocks; // original number of blocks
+   int rblocks; // reduces number of blocks
+   Mesh * mesh = nullptr;
+   bool parallel = false;
+   // original set of Finite Element Spaces
+   Array<FiniteElementSpace *> fes;
+   // indicates if the original space is already a trace space
+   Array<bool> IsTraceSpace;
+
+   // New set of "reduced" Finite Element Spaces
+   // (after static condensation)
+   Array<FiniteElementSpace *> tr_fes;
+
+   Array<int> dof_offsets;
+   Array<int> tdof_offsets;
+
+   Array<int> rdof_offsets;
+   Array<int> rtdof_offsets;
+
+   // Schur complement matrix
+   // S = A_ii - A_ib (A_bb)^{-1} A_bi.
```

Comment:
Isn't it `A_bb - A_bi (A_ii)^{-1} A_ib`?

Reply:
Yeah, here I'm using (b) for "bubble" and (i) for "interface"

Reply:
Oh, I thought (i) for "interior" and (b) for "boundary." The definitions you added definitely help :)